providers/desec/protocol.go

```diff
@@ -58,32 +58,42 @@ type dnssecKey struct {
 type errorResponse struct {
 	Detail string `json:"detail"`
 }
+type nonFieldError struct {
+	Errors []string `json:"non_field_errors"`
+}
 
-func (c *desecProvider) fetchDomainList() error {
-	c.domainIndex = map[string]uint32{}
-	var dr []domainObject
-	endpoint := "/domains/"
-	var bodyString, err = c.get(endpoint, "GET")
+func (c *desecProvider) authenticate() error {
+	endpoint := "/auth/account/"
+	var _, _, err = c.get(endpoint, "GET")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *desecProvider) fetchDomain(domain string) error {
+	endpoint := fmt.Sprintf("/domains/%s", domain)
+	var dr domainObject
+	var bodyString, statuscode, err = c.get(endpoint, "GET")
 	if err != nil {
-		return fmt.Errorf("Failed fetching domain list (deSEC): %s", err)
+		if statuscode == 404 {
+			return nil
+		}
+		return fmt.Errorf("Failed fetching domain: %s", err)
 	}
 	err = json.Unmarshal(bodyString, &dr)
 	if err != nil {
 		return err
 	}
-	for _, domain := range dr {
-		//We store the min ttl in the domain index
-		//This will be used for validation and auto correction
-		c.domainIndex[domain.Name] = domain.MinimumTTL
-	}
+	c.domainIndex[dr.Name] = dr.MinimumTTL
```

Comment:
Please add a comment why dr.MinimumTTL is being used here.

Better yet.... please document in `type desecProvider` what domainIndex represents.

Reply:
I've added some documentation.